### PR TITLE
Updated assignment-1 notebook per requirements

### PR DESCRIPTION
**Pull request by sunshinesharon**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Adding code to inspect the data, preprocess it, and implement KNN for classification using the Wine dataset.

## What did you learn from the changes you have made?
I learned how to standardize data, split it into training and testing sets, and tune hyperparameters using GridSearchCV for better model performance.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
Nope, this is very new to me. The cues helped direct me, I didn't think outside of it. 

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
I initially had issues understanding how to split the data correctly and use GridSearchCV. I reviewed the scikit-learn documentation and followed examples to resolve it.

## How were these changes tested?
The changes were tested by evaluating the accuracy of the KNN model on the test data and ensuring the code ran without errors.

## A reference to a related issue in your repository (if applicable)
None. :) 

## Checklist
- [x] I can confirm that my changes are working as intended
- [x] Created a branch with the correct naming convention.
- [x] Ensured that the repository is public.
- [x] Reviewed the PR description guidelines and adhered to them.
- [x] Verify that the link is accessible in a private browser window.

### CODE REVIEW

**Review by PatelVishakh — CHANGES_REQUESTED** [review]
Assignment 1 Incomplete. Needs the following changes

Comments

Q1)III) The type of variable is categorical, this in a data science setting is asking whether the variable is continuous, integer, categorial. 

Q2)IV) 

You are not meant to create a new dataset. Suppose to use predictors_standardized here 
wine_df = pd.DataFrame({
    'feature1': np.random.rand(100),
    'feature2': np.random.rand(100),
    'class': np.random.randint(0, 2, 100)
})

also test_size is 0.25 not 0.3

Also no output for question 4)
